src/biz/bokhorst/xprivacy/XActivityManager.java

```diff
@@ -80,7 +80,7 @@ public String getClassName() {
 							listHook.add(new XActivityManager(act, null, null));
 						else
 							listHook.add(new XActivityManager(act, PrivacyManager.cSystem, null));
-				} else
+				} else if (!server)
```

Comment:
Yes I was not sure how you wanted this, there are many ways to make it, but a simple argument seamed fine. And later a found a few of your own hooks with a similar argument, so I stuck with it.